docs/source/notes/serialization.rst

```diff
@@ -2,90 +2,337 @@
+lists, and dicts:
+
+::
+
+    >>> d = {'a': torch.tensor([1., 2.]), 'b': torch.tensor([3., 4.])}
+    >>> torch.save(d, 'tensor_dict.pt')
+    >>> torch.load('tensor_dict.pt')
+    {'a': tensor([1., 2.]), 'b': tensor([3., 4.])}
+
+Custom data structures that include PyTorch tensors can also be saved if the
+data structure is pickle-able.
+
+Saving and loading tensors preserves views
 ---------------------------------------------
 
 .. _preserve-storage-sharing:
 
-PyTorch saves the underlying storages so that tensors sharing the same storage before :func:`torch.save`
-will still share storage after :func:`torch.load`.
+Saving tensors preserves their view relationships:
+
+::
+
+    >>> numbers = torch.arange(1, 10)
+    >>> evens = numbers[1::2]
+    >>> torch.save([numbers, evens], 'tensors.pt')
+    >>> loaded_numbers, loaded_evens = torch.load('tensors.pt')
+    >>> loaded_evens *= 2
+    >>> loaded_numbers
+    tensor([ 1,  4,  3,  8,  5, 12,  7, 16,  9])
+
+Behind the scenes, these tensors share the same "storage." See
+`Tensor Views <https://pytorch.org/docs/master/tensor_view.html>`_ for more
+on views and storages.
+
+When PyTorch saves one or more tensors it saves their storages and tensor
+metadata separately. This is an implementation detail that may change in the
+future, but it typically saves space and lets PyTorch easily
+reconstruct the view relationships between the loaded tensors. In the above
+snippet, for example, only a single storage is written to 'tensors.pt'.
+
+In some cases, however, saving storages may be unecessary and create
+prohibitively large files. In the following snippet a storage much larger than
+the saved tensor is written to a file:
 
 ::
 
-    >>> tensor = torch.zeros(1000000)
-    >>> slice1 = tensor[:1000]
-    >>> slice2 = tensor[:10] # slice1 and slice2 share the same storage
-    >>> torch.save([slice1, slice2], 'share.pt')
-    >>> loaded_1, loaded_2 = torch.load('share.pt')
-    >>> loaded_1[0]
-    tensor(0.)
-    >>> loaded_2[0]
-    tensor(0.)
-    >>> loaded_2[0] = 1
-    >>> loaded_1[0] # loaded tensors still share storage
-    tensor(1.)
-
-Note that saving storage instead of tensor itself means the serialized file size might not match tensor size.
-In the example above the whole `tensor`'s storage (of size 1000000) is serialized instead of only slices.
-When tensor is expanded from a smaller storage, serialized file size might be smaller than tensor size as well.
+    >>> large = torch.arange(1, 1000)
+    >>> small = large[0:5]
+    >>> torch.save(small, 'small.pt')
+    >>> loaded_small = torch.load('small.pt')
+    >>> loaded_small.storage().size()
+    999
+
+Instead of saving only the five values in the `small` tensor to 'small.pt,'
+the 999 values in the storage shared with `large` were saved and loaded.
+
+When saving tensors smaller than their storage(s), the size of the saved file
+can be reduced by first cloning the tensors to be saved. Cloning a tensor
```

Comment:
I don't know what "containing only its values" means.  I'd probably frontload the description below, i.e. say something like: "Cloning works to reduce the serialized because a cloned tensor does not preserve view relationships..."

Reply:
I agree with you that "containing only its value" is unclear. I'll improve the clarity.